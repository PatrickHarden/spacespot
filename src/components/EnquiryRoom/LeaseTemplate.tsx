import React from 'react'
import styled from 'styled-components'
import { useIntl } from 'react-intl'
import { useSelector, useDispatch } from 'react-redux'

import userSelectors from 'services/user/selectors'
import actions from 'services/enquiry/actions'
import selectors from 'services/enquiry/selectors'

import FileUpload from 'components/EnquiryRoom/FileUpload'
import upload from 'assets/icons/upload.svg'
import { FileItem } from 'services/global/types'
import { DOC_TYPE_TEMPLATE } from 'services/enquiry'

const TemplateBlock = styled.div`
  margin-bottom: 20px;
`

const UploadButton = styled.button`
  display: flex;
  border: none;
  cursor: pointer;
  background: white;
  color: #404042;
  font-size: 14px;
  font-weight: 550;
  line-height: 18px;
  img {
    margin-right: 8px;
  }
`

const DownloadButton = styled.button`
  border: none;
  padding: 0;
  cursor: pointer;
  background: white;
  color: #6cb9d5;
  font-weight: 500;
  font-size: 15px;
`
const ProvisionUploadFormat = styled.span`
  font-weight: normal;
  margin-left: 10px;
`
const LeaseTemplate = (props: {
  className?: string
  disabled: boolean
  flex: boolean
}) => {
  const { className, disabled, flex } = props
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const dispatch = useDispatch()
  const isLandlord = useSelector(userSelectors.isLandlord)
  const showUpload = isLandlord && !disabled
  const leaseTemplate = useSelector(selectors.getLeaseTemplate)
  const fileName = leaseTemplate
    ? leaseTemplate.fileName
    : flex
    ? ''
    : t('SPECIAL_PROV_TEMPLATE_STANDARD')

  const addDoc = (file: File, reader: FileReader) => (): void => {
    const item: FileItem = {
      key: 0,
      uri: reader.result as string,
      name: file.name,
      type: file.type,
    }
    dispatch(actions.putDocType(item, DOC_TYPE_TEMPLATE))
  }

  const getDoc = () => {
    if (leaseTemplate) {
      dispatch(actions.getDoc(leaseTemplate))
    } else {
      dispatch(actions.getStdTemplate())
    }
  }

  return (
    <TemplateBlock className={className}>
      <p>
        <span>{`${t('SPECIAL_PROV_TEMPLATE_MSG')}: `}</span>
        { fileName &&  
          <DownloadButton data-testid="download" onClick={getDoc}>
            {fileName}
          </DownloadButton>
        }
        { !fileName && 
          <span>{`${t('SPECIAL_PROV_TEMPLATE_MSG_NO_FILE')}`}</span>
        }
      </p>
      {showUpload && (
        <FileUpload accept="application/pdf" addFile={addDoc}>
          <UploadButton data-testid="upload">
            <img src={upload} alt="upload" />
            {flex
              ? t('SPECIAL_PROV_TEMPLATE_UPLOAD_FLEX')
              : t('SPECIAL_PROV_TEMPLATE_UPLOAD')}{' '}
            <ProvisionUploadFormat>
              {t('SPECIAL_PROVISION_UPLOAD_FORMAT')}
            </ProvisionUploadFormat>
          </UploadButton>
        </FileUpload>
      )}
    </TemplateBlock>
  )
}

export default LeaseTemplate
