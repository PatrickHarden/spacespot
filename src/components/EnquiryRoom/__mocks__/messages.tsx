export const messages = [
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message',
    sentAt: '2018-10-18T12:40:35.886+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message1',
    sentAt: '2018-10-18T12:40:43.547+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message2',
    sentAt: '2018-10-18T12:40:44.946+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message3',
    sentAt: '2018-10-18T12:40:45.678+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message4',
    sentAt: '2018-10-18T12:40:46.856+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message5',
    sentAt: '2018-10-23T14:17:59.662+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'jk',
    sentAt: '2018-10-28T10:13:59.805+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message6',
    sentAt: '2018-10-28T10:16:16.585+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '11:44',
    sentAt: '2018-10-28T10:45:02.592+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:34',
    sentAt: '2018-10-28T11:23:27.467+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'll',
    sentAt: '2018-10-28T11:29:37.057+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:30',
    sentAt: '2018-10-28T11:30:13.643+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'The message',
    sentAt: '2018-10-29T08:51:21.007+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Another',
    sentAt: '2018-10-29T09:28:49.589+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Other',
    sentAt: '2018-10-29T09:29:02.900+0000',
  },
]
