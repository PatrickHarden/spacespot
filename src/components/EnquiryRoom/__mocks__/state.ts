export const mockState: any = {
  enquiry: {
    data: {
      spaceId: 'SS_SPACE_665',
      peer1Id: '1',
      peer2Id: '2',
      chatId: 159,
      enquireStatus: 'ACCEPTED',
      negotiationId: 139,
      enquireId: 159,
      flex: false,
    },
  },
  negotiation: {
    data: {
      flexible: false,
      terms: [
        {
          termId: 5,
          value: 'MONTHLY',
        },
        {
          termId: 6,
          value: '2000',
        },
        {
          termId: 7,
          value: '2',
        },
        {
          termId: 8,
          value: '2000',
        },
        {
          termId: 9,
          value: 'GUARANTEE',
        },
        {
          termId: 1,
          value: '3500',
        },
        {
          termId: 2,
          value: '40000',
        },
        {
          termId: 3,
          value: '2020-04-22',
        },
        {
          termId: 4,
          value: '13',
        },
      ],
      negotiationId: 139,
      enquireId: 159,
      createdAt: '2020-02-14T06:30:08.354+0000',
      lastUpdated: '2020-03-05T16:35:02.513+0000',
      lastModifyBy: '0062fb00-e389-4bb8-9119-613e3c45938a',
      status: 'PROPOSED',
      customFitout: false,
      fitoutOption: 0,
      specialProvision: 'gfd',
      specialProvisionStatus: 'CREATED',
    },
    loading: false,
    termsFields: {
      '1': 'Area',
      '2': 'Rent',
      '3': 'Start',
      '4': 'Duration',
      '5': 'Frequency',
      '6': 'AdditionalCosts',
      '7': 'Deposit',
      '8': 'ServiceCharges',
      '9': 'DepositType',
    },
    signerInfo: {
      id: 1,
      name: 'Name1',
      emailId: 'name1@email.com',
      companyName: 'companyA',
      companyNumber: '12345678',
    },
    customFitout: [],
  },
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
