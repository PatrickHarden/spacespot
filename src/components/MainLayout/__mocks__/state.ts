export const mockState: any = {
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
