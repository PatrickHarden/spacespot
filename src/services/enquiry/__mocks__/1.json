{
  "chatId": 1,
  "enquireId": 1,
  "spaceId": 2,
  "peer1Id": "8fa47f17-c44e-48eb-8e88-9e8c8fe5cda3",
  "peer2Id": "8fa47f17-c44e-48eb-8e88-9e8c8fe5cda4"
}
