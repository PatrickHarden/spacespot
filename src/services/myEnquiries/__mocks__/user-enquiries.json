[
  {
    "enquireId": 167,
    "negotiationId": 147,
    "enquireStatus": "PENDING_TO_ACCEPT",
    "spaceId": "SS_SPACE_650",
    "terms": [
      { "termId": 1, "value": "12" },
      { "termId": 2, "value": "12.0" },
      {
        "termId": 3,
        "value": "Tue Feb 25 2020 17:59:47 GMT+0100 (hora estándar de Europa central)"
      },
      { "termId": 4, "value": "123" },
      { "termId": 5, "value": "" },
      { "termId": 6, "value": "" },
      { "termId": 7, "value": "" },
      { "termId": 8, "value": "123.0" },
      { "termId": 9, "value": "" }
    ],
    "messages": [
      {
        "userId": "de2563f1-2e15-4a6a-9e39-9600a8ad4ec1",
        "message": "",
        "sentAt": "2020-02-25T16:59:53.122+0000"
      }
    ]
  },
  {
    "enquireId": 168,
    "negotiationId": 148,
    "enquireStatus": "PENDING_TO_ACCEPT",
    "spaceId": "SS_SPACE_650",
    "terms": [
      { "termId": 1, "value": "12" },
      { "termId": 2, "value": "12.0" },
      {
        "termId": 3,
        "value": "Tue Feb 25 2020 18:00:28 GMT+0100 (hora estándar de Europa central)"
      },
      { "termId": 4, "value": "13" },
      { "termId": 5, "value": "" },
      { "termId": 6, "value": "" },
      { "termId": 7, "value": "" },
      { "termId": 8, "value": "123.0" },
      { "termId": 9, "value": "" }
    ],
    "messages": [
      {
        "userId": "de2563f1-2e15-4a6a-9e39-9600a8ad4ec1",
        "message": "",
        "sentAt": "2020-02-25T17:00:34.014+0000"
      }
    ]
  }
]
