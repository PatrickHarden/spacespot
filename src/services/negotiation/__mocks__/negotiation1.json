{
  "terms": [
    { "termId": 1, "value": "34" },
    { "termId": 2, "value": "207" },
    { "termId": 3, "value": "29 Months" },
    { "termId": 4, "value": "200 Sq Mtr" },
    { "termId": 5, "value": "$5600 per month" },
    { "termId": 6, "value": "20th September" }
  ],
  "negotiationId": 129,
  "enquireId": 149,
  "createdAt": "2020-02-03T13:03:07.661+0000",
  "lastUpdated": "2020-02-11T09:06:15.764+0000",
  "lastModifyBy": "c700793a-1963-4b2b-996f-1939f853cc10",
  "status": "CREATED"
}
