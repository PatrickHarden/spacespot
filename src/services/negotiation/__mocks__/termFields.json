{
  "1": "Area",
  "2": "Rent",
  "3": "Start",
  "4": "Duration",
  "5": "Frequency",
  "6": "AdditionalCosts",
  "7": "Deposit",
  "8": "ServiceCharges"
}
