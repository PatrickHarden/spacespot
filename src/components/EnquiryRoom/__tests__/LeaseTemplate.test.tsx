import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'

import LeaseTemplate from '../LeaseTemplate'
import { mockState } from '../__mocks__/state'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
  createIntlCache: () => jest.fn(),
  createIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDispatch = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}))

it('Renders template link', () => {
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <LeaseTemplate disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByTestId('download'))
  fireEvent.click(wrapper.getByTestId('upload'))
})
