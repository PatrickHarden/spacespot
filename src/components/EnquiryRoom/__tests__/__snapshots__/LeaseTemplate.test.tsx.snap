// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders template link 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa fAeyvH"
  >
    <p>
      <span>
        SPECIAL_PROV_TEMPLATE_MSG: 
      </span>
      <button
        class="sc-htpNat jyOkpE"
        data-testid="download"
      >
        SPECIAL_PROV_TEMPLATE_STANDARD
      </button>
    </p>
    <div>
      <div
        class=""
        tabindex="0"
      >
        <input
          accept="application/pdf"
          autocomplete="off"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
        <button
          class="sc-bwzfXH jNiJCK"
          data-testid="upload"
        >
          <img
            alt="upload"
            src="upload.svg"
          />
          SPECIAL_PROV_TEMPLATE_UPLOAD
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
